File: src/main/java/lu/kolja/appdyenamics/AppDyenamics.java
package lu.kolja.appdyenamics;

import net.minecraft.resources.ResourceLocation;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;

/**
 * Main mod class for AppDyenamics.
 */
@Mod(AppDyenamics.ID)
public class AppDyenamics {
    public static final String ID = "appdyenamics";

    public AppDyenamics() {
        // Register our item registry on the mod event bus
        ModItems.ITEMS.register(FMLJavaModLoadingContext.get().getModEventBus());
    }

    /**
     * Utility to create ResourceLocations within this mod's namespace.
     */
    public static ResourceLocation makeId(String path) {
        return new ResourceLocation(ID, path);
    }
}

File: src/main/kotlin/lu/kolja/appdyenamics/AppDyenamics.kt
package lu.kolja.appdyenamics

import net.minecraft.resources.ResourceLocation
import net.neoforged.fml.common.Mod

@Mod(AppDyenamics.ID)
object AppDyenamics {
    const val ID = "appdyenamics"

    fun makeId(path: String) = ResourceLocation.fromNamespaceAndPath(ID, path)

    init {
    }
}
